File: backend/app/schemas/governance.py
# /app/schemas/governance.py
# SentinelOps – Governance API Schemas

from pydantic import BaseModel
from typing import Optional, Literal
from datetime import datetime


# -------------------------------------------------
# Request: Trigger Governance Run
# -------------------------------------------------
class GovernanceRunRequest(BaseModel):
    repo_id: int
    mode: Literal["github", "local"] = "github"
    ref: Optional[str] = "main"


# -------------------------------------------------
# Response: Governance Run Triggered
# -------------------------------------------------
class GovernanceRunResponse(BaseModel):
    run_id: int
    status: str
    triggered_at: datetime


# -------------------------------------------------
# Response: Latest Governance Posture
# -------------------------------------------------
class LatestGovernanceResponse(BaseModel):
    run_id: int
    repo_id: int
    status: str
    score: int
    violations_count: int
    triggered_at: datetime
    completed_at: Optional[datetime]


# -------------------------------------------------
# Response: Workflow Status (for /workflows route)
# -------------------------------------------------
class WorkflowStatus(BaseModel):
    workflow_id: int
    repo_id: int
    path: str
    status: str
    violations_count: int
    last_validated_at: Optional[datetime]

File: frontend/unified-frontend/src/api/governance.ts
export interface GovernanceRunHistoryItem {
  run_id: number;
  status: string;
  score: number;
  violations_count: number;
  triggered_at: string;
  completed_at: string | null;
}

export interface GovernanceRunHistoryResponse {
  repo_id: number;
  runs: GovernanceRunHistoryItem[];
}

export async function fetchGovernanceHistory(repoId: number): Promise<GovernanceRunHistoryResponse> {
  const res = await fetch(`/api/governance/history?repo_id=${repoId}`);

  if (!res.ok) {
    throw new Error(`Failed to fetch governance history: ${res.status}`);
  }

  return res.json();
}
